docs/Tutorials/Advanced/ticketing/user-code.md

```diff
@@ -0,0 +1,158 @@
+:::
+
+Recall, from the [introduction](introduction.md) that your code had the following outline.
+
+```bash
+/my-ticket-app
+├── contract
+│   └── ...
+├── frontend
+│   └── components
+│   │    └── scanner.js
+│   │    └── qrcode.js
+│   └── state
+│   │    └── App.js
+│   │    └── keyInfo.js
+│   └── utils
+│   │    └── configurations.js
+│   │    └── createTickDrop.js
+│   └── node_modules
+│   │    └── keypom-js
+│   │    └── qrcode.react
+│   │    └── react-zxing
+│   │    └── react
+│   │    └── react-dom
+│   │    └── react-router-dom
+│   │    └── ...
+│   └── package.json
+│   └── package-lock.json
+├── ...
+├── package.json
+├── package-lock.json
+```
+
+This tutorial will be covering the code in `app.js`, `KeyInfo.js`, and `qrcode.js`.
+
+---
+
+## `App.js`
+### Setting Up
+The primary purpose of `app.js` is to display the different states of the claim page. This will involve getting the current key uses, and then rendering based on the value returned. 
+
+The first step is to initialize a connection to NEAR and setup all the state variables that will be needed to render the page. 
+
+```jsx reference showLineNumbers
+https://github.com/keypom/keypom-js/blob/0035852580c8ce848571d89f31ae47a3794414d6/docs-advanced-tutorials/ticket-app/frontend/state/App.js#L15-L59
+```
+
+When the page is loaded, the function `setup` is called and the URL is parsed for `contractId` and `privateKey`. These will be stored in their own respective state  variables (lines 40-45) for further use. Note that the URL is split by `/` and the index of `contractId` and `privateKey` are known. When adapting this code for your own app, you may need to change those index values accordingly.
+
+Once the URL has been parsed and the resulting values stored, `connectNear` is called. This function handles the NEAR connection, as well as the Keypom connection and the linkdrop URL.
+
+:::note 
+When connecting to NEAR, a `BrowserLocalStorageKeyStore` is used rather than an `UnencryptedFileSystemKeyStore`. This will create a key store in the user's browser for future use. For more on key stores, see [here](https://docs.near.org/tools/near-api-js/quick-reference#key-store)
+:::
+
+With the NEAR connection established, your browser can now talk to the NEAR blockchain. The next step, is to call `initKeypom`. This initializes the SDK to allow for interactions with the Keypom smart contracts. Without it, none of the other SDK functions would work as expected. More info on the `initKeypom` function can be found [here](../../../keypom-sdk/modules#initkeypom).
+
+After the Keypom initialization is complete, the SDK function [`formatLinkdropUrl`](../../../keypom-sdk/modules.md#formatlinkdropurl) can be used to create the linkdrop link to embed in the QR code. To use this function, it simply needs a base URL, as well as the `privateKey` to be used. The returned linkdrop link is set to the state variable `link` and will be used to render the QR code.
```

Comment:
why don't we just take the contract ID and private key and turn that into the QR code? Why use format linkdrop URL?